tests/utils.py

```diff
@@ -0,0 +1,62 @@
+from os import path
+from typing import Union
+
+import boa
+from boa.vyper.contract import VyperContract
+from eth.codecs.abi.exceptions import DecodeError as ABIDecodeError
+from eth_account.signers.local import LocalAccount
+
+from scripts.constants import BASE_DIR, ZERO_ADDRESS
+from scripts.deployment_utils import get_deployed_contract
+
+
+def get_contract_pools(contract_name: str, address: str) -> list[str]:
+    """
+    Retrieves the list of pools from a deployed contract with the given address.
+    :param contract_name: The name of the contract to load.
+    :param address: The address of the deployed contract.
+    """
+    registry = get_deployed_contract(contract_name, address)
+    return [registry.pool_list(i) for i in range(registry.pool_count())]
+
+
+def deploy_contract(
+    contract: str,
+    *args,
+    sender: Union[LocalAccount, str],
+    directory: str = ".",
+    **kwargs,
+) -> VyperContract:
+    file_name = path.join(
+        BASE_DIR, f"contracts/mainnet/{directory}/{contract}.vy"
+    )
+    with boa.env.sender(sender):
+        return boa.load(file_name, *args, **kwargs)
+
+
+def assert_decode_error(e: ABIDecodeError):
+    """
+    Checks that the error message is the expected decode error.
+    TODO: This happens in some pools, but it's not clear if it's a boa or contract issue.
```

Comment:
@bout3fiddy do you think this decode issue is worth spending some time on?

Reply:
no we can come back to it later.